src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -61,25 +65,33 @@ public T set(T value, int index) {
     public T remove(int index) {
         checkIndex(index);
         Node<T> currentNode = findCurrentNode(index);
-        currentNode.prev.next = currentNode.next;
-        currentNode.next.prev = currentNode.prev;
-        currentNode.prev = currentNode.next = null;
+        if (size == 1) {
+            currentNode.value = null;
+        } else if (currentNode == head || currentNode != null && currentNode.equals(head)) {
+            currentNode.next.prev = null;
+            head = currentNode.next;
+        } else if (currentNode == tail || currentNode != null && currentNode.equals(tail)) {
+            currentNode.prev.next = null;
+            tail = currentNode.prev;
+        } else {
+            currentNode.prev.next = currentNode.next;
+            currentNode.next.prev = currentNode.prev;
+            currentNode.prev = currentNode.next = null;
+        }
```

Comment:
think of it this way: you can find node on index, and previous to it. between them you have to insert new node. all you need to check - is previous node null or not